contribute/code.rst

```diff
@@ -0,0 +1,211 @@
+
+Write commit messages that answer the Five Ws
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+"The Five Ws" are a set of questions that can help someone gain the full context of a situation quickly. They are "Who?" "What?" "Where?" "When?" and "Why?"
+
+You should strive to answer these questions in your commit metadata or content. Doing so helps reviewers and future readers understand the context of a change. In order of importance, answer:
+
+What?
+    This question should be answered by your commit summary and the first few lines of the commit message. "What does your commit do?" "`Optimize wallpapers for load times and memory use <https://github.com/ubports/unity8/commit/a81421ac1fe5135b9cff710c3cd819aa1804c6e6>`_". Many people stop their commit messages at this point. Continue on...
+Why?
+    You should always state the reasoning behind a change: the bugs it fixes, the features it builds up to, and especially the limitations that cause it to look more complicated than a viewer expects. Provide permanent links to issue reports or documentation if they give more context.
+Who?
+    Knowing who to ask about a change is always important. Obviously, Git will embed your name and email address as the committer of your changes. If someone else has helped you write your changes, you should add them with a ``Co-Authored-By: Their Name <their-email@example.com>`` line in your commit message.
+Where?
+    This is a tricky one. Usually the location in the world where you wrote a commit is irrelevant. The location in the code where the change exists is well-described by the commit data. However, you may want to describe the code surrounding your change if it helps to explain why your change is needed. For example, give some details about an interesting API you're using or bugs in other files or projects that you're working around.
+When?
+    Knowing the events surrounding a change's authoring can be just as important as knowing the date of authorship. Git will automatically embed the date you created your commit into the commit itself. It's up to you to provide any temporal context, such as being asked to author a change during an in-person event or a difficult Agile-style sprint.
+
+Some people also add "How?" to their list of Ws, but the content of your commit rather than its message should answer that question.
+
+It is possible to provide some of this context in code comments rather than in commit messages. How you do it is up to you, but remember that comments can easily become out-of-date compared to the code that they are near. A commit message is linked only to the code you wrote.
+
```

Comment:
It is possible to provide some of this context in code comments rather than in commit messages. Just bear in mind that comments can easily become out-of-date compared to the code that they are near. A commit message is linked only to the code you wrote. On the other hand some information need to be given to future developers who only look at the code but do not check the commit history.